kodilla-stream/src/main/java/com/kodilla/stream/StreamMain.java

```diff
@@ -1,57 +1,26 @@
 package com.kodilla.stream;
 
-import com.kodilla.stream.beautifier.PoemBeautifier;
-import com.kodilla.stream.beautifier.PoemDecorator;
-import com.kodilla.stream.lambda.*;
-import com.kodilla.stream.reference.FunctionalCalculator;
+import com.kodilla.stream.forumuser.Forum;
+import com.kodilla.stream.forumuser.ForumUser;
 
-import java.sql.SQLOutput;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 public class StreamMain {
     public static void main(String[] args) {
 
-        //solution 1 - OOP
-        //SaySomething saySomething = new SaySomething();
-        //saySomething.say();
+        Forum forumStream = new Forum();
 
-        //solution 2 - interface
-        Processor processor = new Processor();
-        //ExecuteSaySomething executeSaySomething = new ExecuteSaySomething();
-        //processor.execute(executeSaySomething);
+        Map<Integer, ForumUser> usersByIndividualNumber = forumStream.getUserList().stream()
+                .filter(ForumUser -> ForumUser.getGender() == 'M')
+                .filter(ForumUser -> ForumUser.getYear() <= 2000)
+                .filter(ForumUser -> ForumUser.getNoOfPosts() >= 1)
+                .collect(Collectors.toMap(ForumUser::getUserIndividualNumber, ForumUser -> ForumUser));
```

Comment:
Ciekawostka: zamiast pisać `x -> x`, możesz użyć referencji na funkcję tożsamościową: https://docs.oracle.com/javase/8/docs/api/java/util/function/Function.html#identity--

Reply:
ok. Zmienione.